qiskit/pulse/instructions/play.py

```diff
@@ -46,20 +47,34 @@ def __init__(self, pulse: Pulse,
         """
         if not isinstance(pulse, Pulse):
             raise PulseError("The `pulse` argument to `Play` must be of type `library.Pulse`.")
-        self._pulse = pulse
-        self._channel = channel
         if name is None:
             name = pulse.name
         super().__init__((pulse, channel), pulse.duration, (channel,), name=name)
 
     @property
     def pulse(self) -> Pulse:
         """A description of the samples that will be played."""
-        return self._pulse
+        return self.operands[0]
 
     @property
     def channel(self) -> PulseChannel:
         """Return the :py:class:`~qiskit.pulse.channels.Channel` that this instruction is
         scheduled on.
         """
-        return self._channel
+        return self.operands[1]
+
+    def assign_parameters(self,
+                          value_dict: Dict[ParameterExpression,
+                                           Union[ParameterExpression, int, float, complex]],
+                          inplace: bool = True) -> 'Play':
+        play_inst = super().assign_parameters(value_dict, inplace)
+
+        if hasattr(self.pulse, 'assign_parameters'):
```

Comment:
Why is this required? Why not a standard method on a pulse and have it do nothing for a waveform?

Reply:
Okay, that makes sense